src/Validation/Validator.php

```diff
@@ -314,14 +321,40 @@ public function add($field, $name, $rule = [])
             $rules = $name;
         }
 
+        if ($this->_autoRequire) {
+            $rules += [
+                '_requirePresence' => []
+            ];
+
+            $rules['_requirePresence'] += [
+                'mode' => true,
+                'message' => null
+            ];
+
+            $this->requirePresence($field, $rules['_requirePresence']['mode'], $rules['_requirePresence']['message']);
+        }
+        unset($rules['_requirePresence']);
+
         foreach ($rules as $name => $rule) {
-            $field->add($name, $rule);
+            $ruleSet->add($name, $rule);
         }
 
         return $this;
     }
 
     /**
+     * Adds automaticly requirePresence on each field added through add method
+     *
+     * @param bool $require whether auto require is on or off
+     * @return $this
+     */
+    public function autoRequire($require)
```

Comment:
yes thats the concept behind it, i have form with 10 mandatory fields and 2 optional, so instead of writing 10x requirePresence i came up with this solution